typegen/typescript.printer.go

```diff
@@ -124,24 +127,38 @@ func PrintTsTypes(parser *Parser, w io.Writer, stringify Stringifier) {
 		return w.String()
 	}
 
+	enumTemplate, err := template.New("enum template").Funcs(template.FuncMap{
+		"Serialize": func(t IType) string {
+			switch v := t.(type) {
+			case *EnumDef:
+				res := fmt.Sprintf("export enum %sEnum {\n", t.GetName())
+				for _, v := range v.Values {
+					res += fmt.Sprintf("    %s = %s,\n", v.name, v.Stringify())
+				}
+				res += "}\n"
+				return res
```

Comment:
Can you do it as a part of template itself?
It looks you can even check type with builtin means:
https://discourse.gohugo.io/t/how-to-test-if-a-variable-is-of-a-certain-type/9207/2